File: sim_backend/src/main/java/sim/data/domain/SimcardSearchModel.java
package sim.data.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class SimcardSearchModel {

    private Integer simcardid;
    private String phonenumber;
    private Integer grupaid;
    private Integer filialid;
    @JsonFormat(pattern = "dd-MM-yyyy",timezone = "Europe/Kiev")
    private Date periodbegin;
    @JsonFormat(pattern = "dd-MM-yyyy",timezone = "Europe/Kiev")
    private Date periodend;
    private Integer officialrankid;
    private Integer tariffnametarifid;
    private Integer addressid;
    private String du;
    private String mp;
    private String note1;
}

File: sim_frontend/src/app/models/simcard.search.model.ts
import {FilialModel} from "./filial.model";
import {GrupaModel} from "./grupa.model";
import {OfficialrankModel} from "./officialrank.model";

export class SimcardSearchModel{
    constructor(
        public simcardid? :number,
        public phonenumber? :string,
        public grupaid? :number,
        public filialid? :number,
        public periodbegin?: string,
        public periodend?: string,
        public officialrankid? :number,
        public tariffnametarifid? :number,
        public addressid? :number,
        public du? :string,
        public mp? :string,
        public note1? :string,
    ) {}
}
